### PR TITLE
feat: implement PREVRANDAO opcode

### LINKED ISSUE #50: Opcode: PREVRANDAO

### PR DESCRIPTION
**Pull request by ricomateo**
Closes #50 
This PR implements code generation for [PREVRANDAO](https://www.evm.codes/#44?fork=cancun) opcode.

### CODE REVIEW

**Comment by JulianVentura** [comment]
LGTM

File: src/syscall.rs [hunk 1]
```diff
@@ -311,6 +311,14 @@ impl<'c> SyscallContext<'c> {
         let log = Log { data, topics };
         self.inner_context.logs.push(log);
     }
+
+    #[allow(improper_ctypes)]
+    pub extern "C" fn get_prevrandao(&self, prevrandao: &mut U256) {
+        if let Some(randao) = self.env.block.prevrandao {
+            prevrandao.hi = (randao >> 128).low_u128();
+            prevrandao.lo = randao.low_u128();
+        }
```
Comment (MegaRedHand) [inline src/syscall.rs:320]: We should set it to zero when randao isn't set:
```rust
let randao = self.env.block.prevrandao.unwrap_or_default();
prevrandao = ...;
```

**Review by MegaRedHand — CHANGES_REQUESTED** [review]

File: tests/evm.rs [hunk 2]
```diff
@@ -8,7 +10,7 @@ use evm_mlir::{
     Env, Evm,
 };
 use num_bigint::BigUint;
-use std::str::FromStr;
+type B256 = ethereum_types::U256;
```
Comment (MegaRedHand) [inline tests/evm.rs:13]: This shouldn't be here

File: tests/evm.rs [hunk 3]
```diff
@@ -83,6 +85,34 @@ fn run_program_assert_gas_exact(operations: Vec<Operation>, env: Env, needed_gas
     assert!(result.is_halt());
 }
 
+fn run_program_assert_bytes_result(
```
Comment (MegaRedHand) [inline tests/evm.rs:88]: There're new functions in main to do this

**Review by MegaRedHand — COMMENTED** [review]
Please fix conflicts

File: src/codegen/context.rs [hunk 4]
```diff
@@ -874,6 +874,22 @@ impl<'c> OperationCtx<'c> {
         )
     }
 
+    #[allow(unused)]
```
Comment (MegaRedHand) [inline src/codegen/context.rs:877]: This isn't needed

File: src/program.rs [hunk 5]
```diff
@@ -54,6 +54,7 @@ pub enum Opcode {
     COINBASE = 0x41,
     TIMESTAMP = 0x42,
     NUMBER = 0x43,
+    PREVRANDAO = 0x44,
     // DIFFICULTY = 0x44,
```
Comment (MegaRedHand) [inline src/program.rs]: We can remove `DIFFICULTY`. It was replaced in https://eips.ethereum.org/EIPS/eip-4399

File: src/env.rs [hunk 6]
```diff
@@ -59,7 +59,7 @@ pub struct BlockEnv {
     // NOTE: `prevrandao` can be found in a block in place of `mix_hash`.
     //
     // [EIP-4399]: https://eips.ethereum.org/EIPS/eip-4399
-    //pub prevrandao: Option<B256>,
+    pub prevrandao: Option<U256>,
```
Comment (MegaRedHand) [inline src/env.rs:62]: This should be a `Option<B256>`

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**